From 54b843af230e6f1feb18e8fa8510d9ddacae4d18 Mon Sep 17 00:00:00 2001
From: Steve Dickson <SteveD@redhat.com>
Date: Thu, 26 Jun 2014 15:54:18 -0400
Subject: [net] sunrpc/xprtrdma: Simplify rpcrdma_deregister_external() synopsis

Message-id: <1403798077-28799-60-git-send-email-steved@redhat.com>
Patchwork-id: 84919
O-Subject: [RHEL7.1 PATCH 59/78] xprtrdma: Simplify rpcrdma_deregister_external() synopsis
Bugzilla: 1113248
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>

From: Chuck Lever <chuck.lever@oracle.com>

Clean up: All remaining callers of rpcrdma_deregister_external()
pass NULL as the last argument, so remove that argument.

Signed-off-by: Chuck Lever <chuck.lever@oracle.com>
Tested-by: Steve Wise <swise@opengridcomputing.com>
Signed-off-by: Anna Schumaker <Anna.Schumaker@Netapp.com>
Upstream Id: 13c9ff8f673862b69e795ea99a237b461c557eb3
Bugzilla: 1113248
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/sunrpc/xprtrdma/rpc_rdma.c b/net/sunrpc/xprtrdma/rpc_rdma.c
index aae1726..436d229 100644
--- a/net/sunrpc/xprtrdma/rpc_rdma.c
+++ b/net/sunrpc/xprtrdma/rpc_rdma.c
@@ -270,7 +270,7 @@ rpcrdma_create_chunks(struct rpc_rqst *rqst, struct xdr_buf *target,
 out:
  for (pos = 0; nchunks--;)
   pos += rpcrdma_deregister_external(
-    &req->rl_segments[pos], r_xprt, NULL);
+    &req->rl_segments[pos], r_xprt);
  return 0;
 }
 
diff --git a/net/sunrpc/xprtrdma/transport.c b/net/sunrpc/xprtrdma/transport.c
index c23b0c1..430cabb 100644
--- a/net/sunrpc/xprtrdma/transport.c
+++ b/net/sunrpc/xprtrdma/transport.c
@@ -575,7 +575,7 @@ xprt_rdma_free(void *buffer)
  for (i = 0; req->rl_nchunks;) {
   --req->rl_nchunks;
   i += rpcrdma_deregister_external(
-   &req->rl_segments[i], r_xprt, NULL);
+   &req->rl_segments[i], r_xprt);
  }
 
  if (req->rl_iov.length == 0) { /* see allocate above */
diff --git a/net/sunrpc/xprtrdma/verbs.c b/net/sunrpc/xprtrdma/verbs.c
index 7c7e9b4..0cbc83c 100644
--- a/net/sunrpc/xprtrdma/verbs.c
+++ b/net/sunrpc/xprtrdma/verbs.c
@@ -1632,7 +1632,7 @@ rpcrdma_register_external(struct rpcrdma_mr_seg *seg,
 
 int
 rpcrdma_deregister_external(struct rpcrdma_mr_seg *seg,
-  struct rpcrdma_xprt *r_xprt, void *r)
+  struct rpcrdma_xprt *r_xprt)
 {
  struct rpcrdma_ia *ia = &r_xprt->rx_ia;
  int nsegs = seg->mr_nsegs, rc;
@@ -1658,12 +1658,6 @@ rpcrdma_deregister_external(struct rpcrdma_mr_seg *seg,
  default:
   break;
  }
- if (r) {
-  struct rpcrdma_rep *rep = r;
-  void (*func)(struct rpcrdma_rep *) = rep->rr_func;
-  rep->rr_func = NULL;
-  func(rep); /* dereg done, callback now */
- }
  return nsegs;
 }
 
diff --git a/net/sunrpc/xprtrdma/xprt_rdma.h b/net/sunrpc/xprtrdma/xprt_rdma.h
index bf08ee0..3f44d6a 100644
--- a/net/sunrpc/xprtrdma/xprt_rdma.h
+++ b/net/sunrpc/xprtrdma/xprt_rdma.h
@@ -331,7 +331,7 @@ int rpcrdma_deregister_internal(struct rpcrdma_ia *,
 int rpcrdma_register_external(struct rpcrdma_mr_seg *,
     int, int, struct rpcrdma_xprt *);
 int rpcrdma_deregister_external(struct rpcrdma_mr_seg *,
-    struct rpcrdma_xprt *, void *);
+    struct rpcrdma_xprt *);
 
 /*
  * RPC/RDMA connection management calls - xprtrdma/rpc_rdma.c
-- 
1.7.1